### PR TITLE
Ignore empty codegen modules in outputs

### PR DESCRIPTION
**Pull request by trevoranderson**
References:
Original report: https://github.com/tokio-rs/prost/issues/228
My report: https://github.com/tokio-rs/prost/issues/724
Partial fix in: https://github.com/tokio-rs/prost/pull/605
Issue with fix: https://github.com/hyperium/tonic/issues/964
Reverted in: https://github.com/tokio-rs/prost/pull/639

After digging in, the issue with the first fix is that while it prevented the file from being generated, it did not prevent the file from being referenced in the "include" file because both locations iterate the module hash. My proposed solution is to instead avoid adding the empty module to the list at all. Then the file does not get output and the include file does not have an entry that would otherwise be referencing an empty file.

Also did a small refactor pass on the surrounding loop to destructure the tuple instead of using tuple.0 syntax wherever it is used.

Tested by reproducing the original issue using `tonic-build` with the original behavior working then reintroduced the bug, then fixed it with my changes. Manually verified output files and included them in a new rust project to make sure they worked and verified that the extra file is not present in file system or the include file. Also ran cargo test on prost

### CODE REVIEW

**Comment by LucioFranco** [comment]
@trevoranderson thank you for doing this! Would it be possible to add a test of the original behavior that was fixed with this so we can make sure we don't regress in the future?

**Comment by trevoranderson** [comment]
Made an attempt at that too. I used an proto adapted from the hyperium issue report which reproduces the issue without my fix and works with the fix applied.

File: prost-build/src/lib.rs [hunk 1]
```diff
@@ -156,6 +150,13 @@ use crate::ident::to_snake;
 use crate::message_graph::MessageGraph;
 use crate::path::PathMap;
 
+mod ast;
```
Comment (LucioFranco) [inline prost-build/src/lib.rs:153]: Why this change?

**Review by LucioFranco — APPROVED** [review]
LGTM thanks!
